webservices/cxf/src/java/org/sakaiproject/webservices/SakaiLTI.java

```diff
@@ -0,0 +1,165 @@
+		){
+		
+		Session session = establishSession(sessionid);
+
+        boolean customTitle = false;
+        
+        String toolid = "sakai.basiclti";
+
+        try {
+
+            //get site
+            Site siteEdit = siteService.getSite(siteId);
+
+            //add the page
+            SitePage sitePageEdit = siteEdit.addPage();
+            sitePageEdit.setTitle(toolTitle);
+            sitePageEdit.setTitleCustom(true);
+            sitePageEdit.setLayout(0);
+            sitePageEdit.setPosition(0);
+            sitePageEdit.setPopup(false);
+
+            // Check that the tool is visible (not stealthed) and available for this site type (category)
+            if (!securityService.isSuperUser(session.getUserId())) {
+
+                Set categories = new HashSet<String>();
+                Set<Tool> visibleTools = toolManager.findTools(categories, null);
+
+                boolean toolVisible = false;
+                for (Tool tool : visibleTools) {
+                    if (tool.getId().equals(toolid)) {
+                        toolVisible = true;
+                    }
+                }
+
+                if (!toolVisible) {
+                    LOG.warn("WS addLTITool(): Permission denied. Must be super user to add a stealthed tool to a site.");
+                    throw new RuntimeException("WS addLTITool(): Permission denied. Must be super user to add a stealthed tool to a site.");
+                }
+
+                categories.add(siteEdit.getType());
+                Set<Tool> availableTools = toolManager.findTools(categories, null);
+
+                boolean toolAvailable = false;
+                for (Tool tool : availableTools) {
+                    if (tool.getId().equals(toolid)) {
+                        toolAvailable = true;
+                    }
+                }
+
+                if (!toolAvailable) {
+                    LOG.warn("WS addLTITool(): Permission denied. Must be super user to add a tool which is not available for this site type.");
+                    throw new RuntimeException("WS addLTITool(): Permission denied. Must be super user to add a tool which is not available for this site type.");
+                }
+            }
+
+            //add the tool
+            ToolConfiguration tool = sitePageEdit.addTool();
+            
+            //set LTI properties
+            setToolProperties(tool, properties);
+
+            tool.setTool(toolid, toolManager.getTool(toolid));
+
+            if (StringUtils.isNotBlank(toolTitle)) {
+                tool.setTitle(toolTitle);
+            } else {
+                tool.setTitle(toolManager.getTool(toolid).getTitle());
+            }
+
+            siteService.save(siteEdit);
+            LOG.info("LTI tool added for site:" + siteId);
+
+            return "success";
+        } catch (Exception e) {
+            LOG.error("WS addToolAndPageToSite(): " + e.getClass().getName() + " : " + e.getMessage());
```

Comment:
I'm not the author of the patch so sorry for the mistake, will correct it and squash the changes in one commit.